cap/modules/repos/serializers.py

```diff
@@ -124,44 +116,55 @@ def get_author(self, obj):
             'id': obj['user_id'],
         }
 
+
+class GitLabPushPayloadSchema(GitLabPayloadSchema):
+    """Serializer for a GitLab webhook payload, from a `push` event."""
+    event_type = fields.Str(default='push', dump_only=True)
+    branch = fields.Method('get_branch', dump_only=True)
+    commit = fields.Method('get_commit', dump_only=True)
+    link = fields.Method('get_link', dump_only=True)
+
     def get_branch(self, obj):
         """Branch information (only for push event)."""
-        if obj['event_name'] == 'push':
-            return obj['ref'].split('/')[-1]
+        return obj['ref'].split('/')[-1]
 
     def get_commit(self, obj):
         """Push-event-specific information."""
-        if obj['event_name'] == 'push':
-            try:
-                return [{
-                    'id': head['id'],
-                    'message': head['message'],
-                    'added': head['added'],
-                    'removed': head['removed'],
-                    'modified': head['modified']
-                } for head in obj['commits']]
-            except KeyError:
-                return
+        try:
+            return [{
+                'id': head['id'],
+                'message': head['message'],
+                'added': head['added'],
+                'removed': head['removed'],
+                'modified': head['modified']
+            } for head in obj['commits']]
+        except KeyError:
+            return
 
     def get_link(self, obj):
         """Link to gitlab page."""
-        if obj['event_name'] == 'tag_push':
-            return f"{obj['project']['homepage']}" \
-                   f"{obj['ref'].replace('refs','')}"
-        else:
-            try:
-                return obj['commits'][0]['url']
-            except (KeyError, IndexError):
-                return
+        try:
+            return obj['commits'][0]['url']
+        except IndexError:
+            return
+
```

Comment:
here you dont need IndexError check (https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#push-events)